tls/ttls/net.c

```diff
@@ -31,547 +32,227 @@
-        }
-    }
-
-    return( 0 );
-}
+	ret = srv_socket->ops->bind(srv_socket, (struct sockaddr*)&sin,
+				    sizeof(sin));
+	if(ret < 0)
+		return MBEDTLS_ERR_NET_BIND_FAILED;
 
-/*
- * Set the socket blocking or non-blocking
- */
-int mbedtls_net_set_block( mbedtls_net_context *ctx )
-{
-#if ( defined(_WIN32) || defined(_WIN32_WCE) ) && !defined(EFIX64) && \
-    !defined(EFI32)
-    u_long n = 0;
-    return( ioctlsocket( ctx->fd, FIONBIO, &n ) );
-#else
-    return( fcntl( ctx->fd, F_SETFL, fcntl( ctx->fd, F_GETFL ) & ~O_NONBLOCK ) );
-#endif
-}
+	ret = srv_socket->ops->listen(srv_socket, 5);
+	if(ret < 0)
+		return MBEDTLS_ERR_NET_LISTEN_FAILED;
 
-int mbedtls_net_set_nonblock( mbedtls_net_context *ctx )
-{
-#if ( defined(_WIN32) || defined(_WIN32_WCE) ) && !defined(EFIX64) && \
-    !defined(EFI32)
-    u_long n = 1;
-    return( ioctlsocket( ctx->fd, FIONBIO, &n ) );
-#else
-    return( fcntl( ctx->fd, F_SETFL, fcntl( ctx->fd, F_GETFL ) | O_NONBLOCK ) );
-#endif
-}
+	ctx->socket = srv_socket;
 
-/*
- * Portable usleep helper
- */
-void mbedtls_net_usleep( unsigned long usec )
-{
-#if defined(_WIN32)
-    Sleep( ( usec + 999 ) / 1000 );
-#else
-    struct timeval tv;
-    tv.tv_sec  = usec / 1000000;
-#if defined(__unix__) || defined(__unix) || \
-    ( defined(__APPLE__) && defined(__MACH__) )
-    tv.tv_usec = (suseconds_t) usec % 1000000;
-#else
-    tv.tv_usec = usec % 1000000;
-#endif
-    select( 0, NULL, NULL, NULL, &tv );
-#endif
+	return 0;
 }
+EXPORT_SYMBOL(mbedtls_net_bind);
 
 /*
- * Read at most 'len' characters
+ * Accept a connection from a remote client
  */
-int mbedtls_net_recv( void *ctx, unsigned char *buf, size_t len )
+int
+mbedtls_net_accept(mbedtls_net_context *bind_ctx,
+		   mbedtls_net_context *client_ctx,
+		   void *client_ip, size_t buf_size,
+		   size_t *ip_len)
 {
-    int ret;
-    int fd = ((mbedtls_net_context *) ctx)->fd;
-
-    if( fd < 0 )
-        return( MBEDTLS_ERR_NET_INVALID_CONTEXT );
-
-    ret = (int) read( fd, buf, len );
-
-    if( ret < 0 )
-    {
-        if( net_would_block( ctx ) != 0 )
-            return( MBEDTLS_ERR_SSL_WANT_READ );
+	int ret;
+	struct socket *srv_socket = bind_ctx->socket;
+	struct socket *cl_socket;
 
-#if ( defined(_WIN32) || defined(_WIN32_WCE) ) && !defined(EFIX64) && \
-    !defined(EFI32)
-        if( WSAGetLastError() == WSAECONNRESET )
-            return( MBEDTLS_ERR_NET_CONN_RESET );
-#else
-        if( errno == EPIPE || errno == ECONNRESET )
-            return( MBEDTLS_ERR_NET_CONN_RESET );
+	ret = sock_create(PF_INET, SOCK_STREAM, IPPROTO_TCP, &cl_socket);
+	if(ret < 0)
+		return MBEDTLS_ERR_NET_ACCEPT_FAILED;
 
-        if( errno == EINTR )
-            return( MBEDTLS_ERR_SSL_WANT_READ );
+	ret = srv_socket->ops->accept(srv_socket, cl_socket, 0);
+	if (ret < 0) {
+		switch(ret) {
+#if defined EAGAIN
+		case -EAGAIN:
+#endif
+#if defined EWOULDBLOCK && EWOULDBLOCK != EAGAIN
+		case -EWOULDBLOCK:
 #endif
```

Comment:
Don't you know what is defined? Usually such ifdefs are used in crossplathform code, not in our case.